calculation/gmhazard_calc/gmhazard_calc/rupture/rupture.py

```diff
@@ -103,3 +109,70 @@ def rupture_id_ix_to_rupture_id(
 ):
     """Converts rupture id ix to rupture ids"""
     return ensemble.get_rupture_ids(rupture_id_ind)
+
+
+def get_fault_header_points(fault: nhm.NHMFault):
+    srf_points = []
+    srf_header: List[Dict[str, Union[int, float]]] = []
+    lon1, lat1 = fault.trace[0]
+    lon2, lat2 = fault.trace[1]
+    strike = geo.ll_bearing(lon1, lat1, lon2, lat2, midpoint=True)
+
+    if 180 > fault.dip_dir - strike >= 0:
+        # If the dipdir is not to the right of the strike, turn the fault around
+        indexes = range(len(fault.trace))
+    else:
+        indexes = range(len(fault.trace) - 1, -1, -1)
```

Comment:
I have a feeling that one of us would mention
```python
indexes = range(len(fault.trace))
if 180 > fault.dip_dir - strike >= 0
else range(len(fault.trace) - 1, -1, -1)
```

Reply:
`reversed(range(len(fault.trace)))` is slightly easier to read imo.

Reply:
Can't quite use reversed to make it easier to read as below in the line where it does 
`for i, i2 in zip(indexes[:-1], indexes[1:]):`
As when you use reversed it creates a range iterator which is not subscriptable where as range is

Reply:
Interesting, tmyk